File: backend/app/schemas/issue.py
from pydantic import BaseModel, Field
from datetime import datetime


class IssueCreate(BaseModel):
    title: str = Field(..., max_length=200)
    description: str | None = None
    assignee_name: str | None = None
    executor_name: str | None = None
    executor_type: str | None = None
    executor_note: str | None = None
    related_files: list[dict] = []
    evidence_note: str | None = None


class IssueUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    assignee_name: str | None = None
    executor_name: str | None = None
    executor_type: str | None = None
    executor_note: str | None = None
    related_files: list[dict] | None = None
    evidence_note: str | None = None
    version: int


class IssueResponse(BaseModel):
    id: int
    workspace_id: int
    title: str
    description: str | None
    status: str
    creator_id: int
    assignee_name: str | None
    executor_name: str | None
    executor_type: str | None
    executor_note: str | None
    related_files: list[dict]
    evidence_note: str | None
    root_cause: str | None
    failed_attempts: str | None
    final_solution: str | None
    reusable: bool
    tags: list[str]
    activity_log: list[dict]
    version: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class StatusChangeRequest(BaseModel):
    status: str
    version: int
    details: str | None = None


class ArchiveRequest(BaseModel):
    root_cause: str
    failed_attempts: str
    final_solution: str
    reusable: bool
    tags: list[str] = []
    version: int

File: frontend/src/types/issue.ts
export type IssueStatus = 'open' | 'in_progress' | 'resolved' | 'archived'
export type ExecutorType = 'human' | 'ai_window' | 'role' | 'tool'

export interface RelatedFile {
  name: string
  path: string
  note: string
}

export interface ActivityLogEntry {
  timestamp: string
  user_id: number
  username: string
  action: string
  details?: string
  from_status?: string
  to_status?: string
}

export interface Issue {
  id: number
  workspace_id: number
  title: string
  description: string | null
  status: IssueStatus
  creator_id: number
  assignee_name: string | null
  executor_name: string | null
  executor_type: ExecutorType | null
  executor_note: string | null
  related_files: RelatedFile[]
  evidence_note: string | null
  root_cause: string | null
  failed_attempts: string | null
  final_solution: string | null
  reusable: boolean
  tags: string[]
  activity_log: ActivityLogEntry[]
  version: number
  created_at: string
  updated_at: string
}

export interface IssueCreate {
  title: string
  description?: string
  assignee_name?: string
  executor_name?: string
  executor_type?: ExecutorType
  executor_note?: string
  related_files?: RelatedFile[]
  evidence_note?: string
}

export interface IssueUpdate {
  title?: string
  description?: string
  assignee_name?: string
  executor_name?: string
  executor_type?: ExecutorType
  executor_note?: string
  related_files?: RelatedFile[]
  evidence_note?: string
  version: number
}

export interface StatusChangeRequest {
  status: IssueStatus
  version: number
  details?: string
}

export interface ArchiveRequest {
  root_cause: string
  failed_attempts: string
  final_solution: string
  reusable: boolean
  tags?: string[]
  version: number
}
